### DISCUSSION TITLE
Anyway to force DNS to utilize IPv4 over IPv6?

### DISCUSSION THREAD

**Discussion by tonedef** [espressif/arduino-esp32#9988] [Q&A]
I spent a lot of time chasing down "DNS failed" issues on an ESP32-S3 board that happen after ~10min after boot. Came across posts like #3722 and #2778 which led me to try disabling IPv6 on my stock router. It "fixed" the issue, but I'm not convinced it's the router's fault as ONLY the S3 has this problem.

This is unacceptable for IoT.  I cannot control other people's networks even if it isn't espressif's fault. If I at least had more control on the ESP32 side, I could opt to fallback to IPv4 to ensure the device works as it should.

**Comment by me-no-dev** [comment]
we are tracking the issue. The problem is that not all V6 enabled routers trigger it. We can not internally reproduce this and users also confirm that it does not happen everywhere. Once we have more clarity what and why it happens, we will add a fix. Issue is somewhere in ESP-IDF's LwIP
